readthedocs/rtd_tests/tests/test_api.py

```diff
@@ -681,6 +681,7 @@ def test_project_environment_variables(self):
             EnvironmentVariable,
             name='TOKEN',
             value='a1b2c3',
+            public=True,
```

Comment:
This could be a problem: You said I should set `public_only` to `True`, which changes behavior of this API. (As the default serializer calls the method without arguments)

If we want to maintain behavior, the `environment_variables` method needs to be made a property of `Version`, where it has the necessary context to infer what variables are visible for that version, or we need to customize the serializer.